src/Persistence/Type/Basic.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\ui\Persistence\Type;
+
+/**
+ * Base type class.
+ */
+class Basic
```

Comment:
This dupliates a lot of data Type, not sure, if this is step into the right direction.

Reply:
The idea was to:
- clearly separate each field type value casting to it's own class, i.e. clearer of what each type value casting does; 
- easily add a new field type without having to change Persistence/UI class;
- override each type class easily to adapt for custom needs;

Reply:
add/override is the only plus, but it strong argument to do it so.

I added some design comments directly